test/integration/Quickstart.kt

```diff
@@ -0,0 +1,193 @@
+import com.vaticle.typedb.client.api.TypeDBTransaction
+import com.vaticle.typedb.studio.Studio
+import com.vaticle.typedb.studio.framework.common.WindowContext
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typeql.lang.TypeQL
+import com.vaticle.typeql.lang.query.TypeQLMatch
+import kotlinx.coroutines.delay
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+/**
+ * Some of these tests use delay!
+ *
+ * The rationale for this is that substituting in stub classes/methods would create a lot of friction from release to
+ * release as the tests would require updating to completely reflect all the internal state that changes with each
+ * function. As a heavily state-driven application, duplicating all of this functionality and accurately verifying that
+ * the duplicate is like-for-like is out of scope.
+ *
+ * The delays are:
+ *  - used only when necessary (some data is travelling between the test and TypeDB)
+ *  - generous with the amount of time for the required action.
+ *
+ * However, this is a source of non-determinism and a better and easier way may emerge.
+ */
+class Quickstart {
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    // This test simulates the carrying out of the instructions found at https://docs.vaticle.com/docs/studio/quickstart
+    @Test
+    fun `Quickstart`() {
+        val schemaString = fileNameToString("test/data/schema_string.tql")
+        val dataString = fileNameToString("test/data/data_string.tql")
+        val queryString = fileNameToString("test/data/query_string.tql")
+
+        runComposeRule(composeRule) {
+            setContent {
+                Studio.MainWindowContent(WindowContext(1000, 500, 0, 0))
+            }
+
+            composeRule.waitForIdle()
+
+            // This opens a dialog box (which we can't see through) so we assert that buttons with that text can be
+            // clicked.
+            composeRule.onAllNodesWithText("Connect to TypeDB").assertAll(hasClickAction())
+
+            StudioState.client.tryConnectToTypeDB(DB_ADDRESS) {}
+            // We wait to connect to TypeDB. This can be slow by default on macOS, so we wait a while.
+            delay(2_500)
+            composeRule.waitForIdle()
+            assertTrue(StudioState.client.isConnected)
+            composeRule.waitForIdle()
+            composeRule.onNodeWithText(DB_ADDRESS).assertExists()
+
+            // Same as connecting to typedb, but we can't see dropdowns either.
+            composeRule.onAllNodesWithText("Select Database").assertAll(hasClickAction())
+
+            try {
+                StudioState.client.tryDeleteDatabase(DB_NAME)
+            }
+            catch (_: Exception) {}
+
+            delay(1_000)
+
+            StudioState.client.tryCreateDatabase(DB_NAME) {}
+            // We wait to create the github database.
+            delay(1_000)
```

Comment:
I am actually a big fan of documenting technical debts in code using comments - however, I think this is on the side of "too many"! The long Javadoc comment on the Quickstart method already goes over why we need to extensively use `delay`, so at this point, it's already clear why we're delaying - and therefore we should delete this _particular_ comment (and some others nearby) as it's something that's already implicit from the code + the other comments.